### PR TITLE
Explainable query flag

### PR DESCRIPTION
**Pull request by adammitchelldev**
## What is the goal of this PR?

We've introduced an explanation cache that is used on the server side to cache and return explanations, but since we didn't previously know if an explanation is required ahead of time, we now give clients the option to specify whether or not the explanation cache is enabled for a given query.

Introducing this new option highlights an issue with client-java gaining too many optional parameters and we expect these parameters to change over time, so we're taking the step to use an Options object instead, which will allow clients to use a simple builder syntax to set various options.

## What are the changes implemented in this PR?

- Added explanation flag option
- Updated to new protocol including explanation option
- Updated other dependencies to fix various issues
- Added an options object to replace the `infer` boolean that now contains multiple options.

### CODE REVIEW

File: GraknClient.java [hunk 1]
```diff
@@ -1073,4 +1081,72 @@ public int hashCode() {
...
+
+        @FunctionalInterface
+        interface FlagConsumer {
+            void accept(boolean flag);
+        }
+    }
+
+    public interface Options {
+        OptionsBuilder DEFAULT = new OptionsImpl();
+        OptionsBuilder NO_INFER = DEFAULT.infer(false);
```
Comment (flyingsilverfin) [inline GraknClient.java:1141]: easier to read without these two extra enums - just use `DEFAULT.explain(...)` and its inverse

File: test/integration/answer/AnswerIT.java [hunk 2]
```diff
@@ -169,4 +167,36 @@ public void writingInAReadTransactionThrows() {
             }
         }
     }
+
+    @Test
```
Comment (flyingsilverfin) [inline test/integration/answer/AnswerIT.java:171]: need tests for the other option too - `explain`

Comment (adammitchelldev) [reply test/integration/answer/AnswerIT.java:174]: Have added

File: GraknClient.java (hunk 1 above)
Comment (adammitchelldev) [reply GraknClient.java]: Removed

**Pull request merged**